src/sysinfo.cc

```diff
@@ -366,6 +366,33 @@ std::vector<CPUInfo::CacheInfo> GetCacheSizes() {
 #endif
 }
 
+std::string GetSystemName() {
+#if defined(BENCHMARK_OS_WINDOWS)
+  std::string str;
+  const unsigned COUNT = MAX_COMPUTERNAME_LENGTH+1;
+  TCHAR  hostname[COUNT];
+  DWORD DWCOUNT = COUNT;
+  if (!GetComputerName(hostname, &DWCOUNT))
+    return std::string("Unable to Get Host Name");
+#ifndef UNICODE
+  str = hostname;
+#else
+  std::wstring wStr = hostname;
+  str = std::string(wStr.begin(), wStr.end());
```

Comment:
Does this work? Is this already done elsewhere in the library?
wchar is bigger than char, what makes sure that this isn't lossy?

Reply:
Using Standard wstring_convert to fix this